modules/uhttp/client/client.go

```diff
@@ -61,11 +64,15 @@ func (c *Client) do(ctx fx.Context, req *http.Request) (resp *http.Response, err
 
 // Get is a context-aware, filter-enabled extension of Get() in http.Client
 func (c *Client) Get(ctx fx.Context, url string) (resp *http.Response, err error) {
+	authctx := authenticate(ctx)
```

Comment:
Why not make this a filter similar to the tracerFilter? Makes it more consistent and cleaner.